core/src/main/java/org/apache/carbondata/core/indexstore/ExtendedBlockletWrapper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.carbondata.core.indexstore;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.hadoop.io.Writable;
+
+/**
+ * Wrapper to write and read the list of extended blocklets over RPC communication.
+ */
+public class ExtendedBlockletWrapper implements Writable {
```

Comment:
`ExtendedBlocklet` and `FineGrainBlocklet` both extends from `Blocklet`. So writing a wrapper class for child class will make it too specific. If you can write a wrapper for `Blocklet` class it can be used for super as well as child classes. Even current implementation is fine but if you get time give a try and check the feasibility of writing a wrapper for super class and use it for child classes